lib/controllers/messages.ts

```diff
@@ -337,6 +335,8 @@ export function CreateMessageHandler(
     // extract the user service
     const userService = userAttributes.service;
 
+    const startRequestTime = Date.now();
```

Comment:
You want to use a [high resolution timer](https://nodejs.org/api/process.html#process_process_hrtime_time) here - `Date.now()` may not be monotonic (i.e. affected by system clock drifts, for instance during NTP updates).